lib/src/Uncertainty/Distribution/TruncatedDistribution.cxx

```diff
@@ -276,6 +276,9 @@ void TruncatedDistribution::computeRange()
     const Scalar probability = distribution_.computeProbability(range);
     if (!(probability > 0.0)) throw InvalidArgumentException(HERE) << "Error: the truncation interval does not contain a non-empty part of the support of the distribution";
     normalizationFactor_ = 1.0 / probability;
+
+    // scale quantile epsilon of the inner distribution
+    distribution_.getImplementation()->setQuantileEpsilon(distribution_.getImplementation()->getQuantileEpsilon() * probability);
```

Comment:
This solution to #1870 seems reasonable to me.

Reply:
On second thought, isn't there a risk of `computeRange` begin called multiple times (say, by `setBounds`) and lowering `quantileEpsilon` every time?

Reply:
Use `ResourceMap.GetAsScalar('Distribution-DefaultQuantileEpsilon')` instead of `distribution_.getImplementation()->getQuantileEpsilon()`